recursion/compiler/examples/logic.rs

```diff
@@ -0,0 +1,24 @@
+use p3_baby_bear::BabyBear;
+use sp1_recursion_compiler::prelude::*;
+
+fn main() {
+    let mut builder = AsmBuilder::<BabyBear>::new();
+
+    let a: Bool = builder.constant(false);
+    let b: Bool = builder.constant(true);
+
+    let and: Bool = builder.uninit();
```

Comment:
uninit is a weird name, maybe we can have something similar to plonky2x where it's just builder.constant and builder.init?